cluster-autoscaler/utils/test/test_utils.go

```diff
@@ -103,6 +104,37 @@ func BuildTestPodWithEphemeralStorage(name string, cpu, mem, ephemeralStorage in
 	return pod
 }
 
+// BuildScheduledTestPod builds a scheduled test pod with a given spec
+func BuildScheduledTestPod(name string, cpu, memory int64, nodeName string) *apiv1.Pod {
+	p := BuildTestPod(name, cpu, memory)
+	p.Spec.NodeName = nodeName
+	return p
+}
+
+// SetStaticPodSpec sets pod spec to make it a static pod
+func SetStaticPodSpec(pod *apiv1.Pod) *apiv1.Pod {
+	pod.Annotations[kube_types.ConfigSourceAnnotationKey] = kube_types.FileSource
+	return pod
+}
+
+// SetMirrorPodSpec sets pod spec to make it a mirror pod
+func SetMirrorPodSpec(pod *apiv1.Pod) *apiv1.Pod {
+	pod.ObjectMeta.Annotations[kube_types.ConfigMirrorAnnotationKey] = "mirror"
+	return pod
+}
+
+// SetDSPodSpec sets pod spec to make it a DS pod
+func SetDSPodSpec(pod *apiv1.Pod) *apiv1.Pod {
+	pod.OwnerReferences = GenerateOwnerReferences("ds", "DaemonSet", "extensions/v1beta1", "api/v1/namespaces/default/daemonsets/ds")
```

Comment:
nit: DaemonSet is in apps/v1, not extensions/v1beta1 for quite a while now. This shouldn't (?) affect the tests, but maybe worth using the right API group?

Reply:
Done